server/src/main/java/org/elasticsearch/http/HttpTracer.java

```diff
@@ -89,6 +90,33 @@ HttpTracer maybeLogRequest(RestRequest restRequest, @Nullable Exception e) {
         return null;
     }
 
+    private void logFullContent(RestRequest restRequest) {
+        try (var stream = HttpBodyTracer.getBodyOutputStream(restRequest.getRequestId(), HttpBodyTracer.Type.REQUEST)) {
+            restRequest.content().writeTo(stream);
+        } catch (Exception e2) {
+            assert false : e2; // no real IO here
+        }
+    }
+
+    private void logStreamContent(RestRequest restRequest) {
+        var stream = HttpBodyTracer.getBodyOutputStream(restRequest.getRequestId(), HttpBodyTracer.Type.REQUEST);
+        restRequest.contentStream().addTracingHandler((chunk, isLast) -> {
+            try {
+                chunk.writeTo(stream);
+            } catch (IOException e) {
+                assert false : e; // no real IO
+            } finally {
+                if (isLast) {
+                    try {
+                        stream.close();
```

Comment:
I believe that this is not sufficient the same way any other request could be interrupted in the middle without end content. 

You can use the `ChunkHandler#close` added in the recent PR.